roles/iam-user/tasks/create_groups.yml

```diff
@@ -1,15 +1,29 @@
 ---
 
-- name: Create developer IAM group
+- name: Create IAM groups
   iam:
     state: present
     iam_type: group
-    name: "{{ iam_user_developer_groupname }}"
+    name: "{{ item.groups }}"
+  with_items: "{{ iam_user_users | default([]) }}"
```

Comment:
if you use the map and unique filters, you can reduce this to a unique list of groups to be created, rather than potentially running the task multiple times for the same group. the user list could have multiple users for the same group